flake.nix

```diff
@@ -0,0 +1,52 @@
+{
+  description = "A flake for s2n-tls";
+
+  inputs.nixpkgs.url = github:NixOS/nixpkgs/nixos-22.11;
+
+  outputs = { self, nix, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system};
+      in rec {
+        packages.s2n-tls = pkgs.stdenv.mkDerivation {
+          src = self;
+          name = "s2n-tls";
+          inherit system;
+          # TODO: override the s2n-tls version
+
+          buildInputs = [
+            pkgs.cmake
+            pkgs.openssl_3
+          ]; # s2n-config has find_dependency LibCrypto
+
+          cmakeFlags = [
+            "-DBUILD_SHARED_LIBS=ON"
+            "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+            "-DS2N_NO_PQ=1" # TODO: set when system like aarch64/mips,etc
+          ];
+
+        };
+        defaultPackage = packages.s2n-tls;
+        packages.s2n-tls-openssl3 = pkgs.s2n-tls.overrideAttrs (finalAttrs: previousAttrs: {
+          doCheck = true;
```

Comment:
What exactly is doCheck triggering?

Reply:
unit tests for the flake; by default they don't get run for a default install.